web/html/src/manager/virtualization/guests/console/MessagePopUp.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from "react";
+import { PopUp } from 'components/popup';
+import { Loading } from 'components/utils/Loading';
+import { Form } from 'components/input/Form';
+import { Password } from 'components/input/Password';
+import { Button } from 'components/buttons';
+import { hideDialog } from 'components/dialog/util';
+
+type Props = {
+  id: string,
+  onSubmit?: () => void,
+  popupState: string,
+  model: any,
+  setModel: (model: any) => void,
+  error: React.ReactNode,
+};
+
+export function MessagePopUp(props: Props) {
+  const buttonValues = {
+    askPassword: {
+      text: t('Submit'),
+    },
+    errors: {
+      text: t('Retry'),
+    },
+  }[props.popupState];
+
+  const popupContent = () => {
+    if (props.popupState === 'wait') {
+      return <Loading text={t('Connecting...')} withBorders={false} />;
+    }
+    if (props.popupState === 'askPassword') {
+      return (
+        <Form model={props.model} className="form-horizontal" onChange={props.setModel}>
+          <Password
+            name="password"
+            label={t('Password')}
+            labelClass="col-md-3"
+            divClass="col-md-6"
+          />
+        </Form>
+      );
+    }
+    return props.error;
+  };
+
+  return (
+    <PopUp
+      id={props.id}
+      hideHeader
+      content={popupContent()}
+      footer={
+        buttonValues && [
+          <Button
+            key="submit"
+            className="btn-primary"
+            text={buttonValues.text}
+            title={buttonValues.text}
+            handler={() => {
+              if (props.onSubmit != null) {
+                props.onSubmit();
```

Comment:
This can be written like `() => props.onSubmit?.()`

Reply:
I wasn't aware of that notation. I'll change since it makes it shorter